### DISCUSSION TITLE
Why can't we use dot-notation on `get()` method ?

### DISCUSSION THREAD

**Discussion by foremtehan** [laravel/framework#38501] [Q&A]
```php
$data = [
    'user' => [
        'name' => 'Alex',
    ]
];

collect($data)->get('user.name'); //null
```

any logic why laravel doesnt allow dot-notation in `get` method ?

**Comment by foremtehan** [comment]
if i pr this does taylor reject it ?

**Reply by dennisprudlo** [reply]
You will never know until you try to PR this. But usually dot-notation is used when there is a fixed structure of the array. It may be the case for you in this situation but this addition would break applications that use `get` to access keys with a dot in the name.

```php
$data = [
    'user' => [
        'firstname' => 'John'
    ],
    'user.lastname' => 'Doe'
];

//
// Without dot-notation access
collect($data)->get('user'); // [ 'firstname' => 'John' ]
collect($data)->get('user.lastname'); // 'Doe'

//
// With dot-notation access
data_get($data, 'user'); // [ 'firstname' => 'John' ]
data_get($data, 'user.lastname'); // null
```

Using `user.lastname` would try to access the `lastname` key in the `user` array and therefore return `null` in this case instead the value of the key `user.lastname`.
